app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/data/ApolloCountryClient.kt

```diff
@@ -0,0 +1,40 @@
+package io.github.viabachelora23michaelkutaibakasper.bprapp.data
+
+import android.util.Log
+import com.apollographql.apollo3.ApolloClient
+import io.github.viabachelora23michaelkutaibakasper.bprapp.CountryByCodeQuery
+import io.github.viabachelora23michaelkutaibakasper.bprapp.ExampleQuery
+import io.github.viabachelora23michaelkutaibakasper.bprapp.domain.CountryClient
+import io.github.viabachelora23michaelkutaibakasper.bprapp.domain.SimpleCountry
+
+class ApolloCountryClient() : CountryClient {
+    override suspend fun getCountries(): List<SimpleCountry> {
+       val apolloClient = ApolloClient.Builder()
+            .serverUrl("https://countries.trevorblades.com/")
+            .build()
+        val response = apolloClient.query(ExampleQuery()).execute()
+        Log.d("ApolloCountryClient", "getCountries: ${response.data?.countries}")
+        return response.data?.countries?.map {
+            SimpleCountry(
+                it.name,
+                it.capital,
+                it.code)
+        } ?: emptyList()
+    }
+
+    override suspend fun getCountry(code: String): SimpleCountry {
+       val apolloClient = ApolloClient.Builder()
+            .serverUrl("https://countries.trevorblades.com/")
```

Comment:
We probably want to be able to pass the URL in the function itself in the actual future client implementations, e.g. getCountry(code: String, url: String), getCountries(url: String) 🚀